### DISCUSSION TITLE
Editor Team Question: Menu Options

### DISCUSSION THREAD

**Discussion by minecraftchloe** [Mojang/minecraft-editor#6] [Q&A]
What else (eg. slash commands) would be helpful to include in the menus?

**Comment by CaptainCooky** [comment]
The Dropdown/Pop-out Menus should activate without the additional click. To clarify, I mean the - **File**>>**Export As** and the same Pop-outs under the **World Options** menu. 

The example below shows that the **Actors** menu should pop out without clicking. Small detail but good for UX :D

![image](https://user-images.githubusercontent.com/88385818/224008079-4152b303-13d8-4322-9f44-31777c9fadc3.png)

**Comment by CaptainCooky** [comment]
I don't know if this is the place for Bugs/Issues, but the following are a few I've come across.

### 1. Some Blocks are not included in the *Block Type* menu.

So far, I've only come across this with Concrete and Concrete powder. 
Ordinarily, if we were to use a command to obtain these blocks, we'd have to input the following: `/give @s concrete 64 {dataId}`.
It may be possible. However, I could not select any Concrete variants other than the default White in the Fill Selection pane. 

This image shows that the list is only populated with the White Variants. I was unable to enter a Data Value for the others.

![image](https://user-images.githubusercontent.com/88385818/224012649-69facbf0-4a3a-4553-9fbd-3ed5a221f4ef.png)

### 2. We need to be able to specify a *Blockstate* when selecting a Block from the *Block Type* menu.

This can be replicated when attempting to fill with *Slabs*. If you choose any of the available Slab blocks and click the **Fill Selection** button, the result will be lower slabs only. Maybe there's a method that I've not found, but being able to specify *Upper Slabs* would be great. Obviously, this feature would be good for all possible *Block Permutations*

### 3. The *Block Type* menu needs to be wider or scroll when the text is larger than the field.

See the image below. Some options are too long, and this can cause confusion about which option we are looking for.

![image](https://user-images.githubusercontent.com/88385818/224015078-c5e12aca-f301-488c-8e19-9736ab9c6583.png)

**Comment by minecraftchloe** [comment]
#21 crosspost - ability to turn off weather cycle

**Comment by minecraftchloe** [comment]
#46 crosspost - night vision/fog toggles

**Comment by cute-catgirl** [comment]
I think it would be nice if we could bind our own custom commands to the menus, or even set keybinds for commands - that way if an option doesn't exist, you could just add it... If not in the default editor it could be an extension

**Reply by dacowan** [reply]
This will be possible using 0.4 -- you'd write a small extension that simply registers a menu item and you can associate it with either Script API typescript, or call a slash command from it.
